openmetadata-service/src/main/java/org/openmetadata/service/resources/teams/UserResource.java

```diff
@@ -878,17 +846,15 @@ public Response generateResetPasswordLink(@Context UriInfo uriInfo, @Valid Email
       throw new BadRequestException("Email is not valid.");
     }
     // send a mail to the User with the Update
-    if (isEmailServiceEnabled) {
-      try {
-        sendPasswordResetLink(
-            uriInfo,
-            registeredUser,
-            EmailUtil.getInstance().getPasswordResetSubject(),
-            EmailUtil.PASSWORD_RESET_TEMPLATE_FILE);
-      } catch (Exception ex) {
-        LOG.error("Error in sending mail for reset password" + ex.getMessage());
-        return Response.status(424).entity(new ErrorMessage(424, EMAIL_SENDING_ISSUE)).build();
-      }
+    try {
+      authHandler.sendPasswordResetLink(
+          uriInfo,
+          registeredUser,
+          EmailUtil.getInstance().getPasswordResetSubject(),
+          EmailUtil.PASSWORD_RESET_TEMPLATE_FILE);
+    } catch (Exception ex) {
```

Comment:
we should catch specific exceptions instead of global. May be there is retires that we can do in sending the email.